src/oic/utils/sdb.py

```diff
@@ -553,6 +553,57 @@ def get_by_uid(self, uid: str) -> List[str]:
             if AuthnEvent.from_json(session["authn_event"]).uid == uid
         ]
 
+    def get_uid_by_sub(self, sub):
```

Comment:
I am not sure these are really needed as we already have methods to return the whole session, so this is just a little helper method. Depends on number of uses.

But if these are really needed/beneficial, they need to be moved to the base class to ensure proper API and should use the already existing `get_by_sub` and `get_by_uid` methods to benefit from optimizations.

Reply:
They are as you say helper method and I think useful because they are used by other parts of the code.
Always prefer to implement in one place if possible.